packages/ui/src/elements/CopyLocaleData/index.tsx

```diff
@@ -0,0 +1,138 @@
+'use client'
+
+import type { MappedComponent } from 'payload'
+
+import React, { useCallback } from 'react'
+import { toast } from 'sonner'
+
+import { useForm, useFormModified } from '../../forms/Form/context.js'
+import { useConfig } from '../../providers/Config/index.js'
+import { RenderComponent } from '../../providers/Config/RenderComponent.js'
+import { useDocumentInfo } from '../../providers/DocumentInfo/index.js'
+import { useLocale } from '../../providers/Locale/index.js'
+import { useTranslation } from '../../providers/Translation/index.js'
+import { Button } from '../Button/index.js'
+import { Popup, PopupList } from '../Popup/index.js'
+import './index.scss'
+
+const baseClass = 'copy-locale-data'
+
+export const DefaultComponent: React.FC = () => {
+  const {
+    config: {
+      localization,
+      routes: { api },
+      serverURL,
+    },
+  } = useConfig()
+  const { code: currentLocale } = useLocale()
+  const { id, collectionSlug, globalSlug } = useDocumentInfo()
+  const { i18n, t } = useTranslation()
+  const { submit } = useForm()
+  const modified = useFormModified()
+
+  const copyLocaleData = useCallback(
+    async ({ from, to }) => {
+      if (modified) {
+        // todo: add translation
+        toast.error('you have unsaved data')
+        return null
+      }
+
+      try {
+        let url
+        let method = 'PATCH'
+
+        if (collectionSlug) {
+          url = `${serverURL}${api}/${collectionSlug}/${id}?locale=${from}`
+        } else if (globalSlug) {
+          url = `${serverURL}${api}/globals/${globalSlug}?locale=${from}`
+          method = 'POST'
+        }
+
+        const response = await fetch(url)
+        if (!response.ok) {
+          throw new Error(`Failed to fetch data: ${response.statusText}`)
```

Comment:
I would use the error from the server so that you don't have to worry about translating this string and the error may be more accurate / helpful than this general error throw you have here.

Reply:
Good call - done!

Reply:
Wait, is there a reason this isn't using `toast.error`? I would expect this: `toast.error(err.message || t('error:unknown'))` and then an early return. I think that is preferred. The `response.statusText` won't have much and because you're throwing an error the frontend will just crash without giving a user friendly error message.

Reply:
I've removed this line completely `throw new Error(`Failed to fetch data: ${response.statusText}`)`, this code is wrapped in a try/catch so now if the response is unsuccessful it will be caught with that instead of throwing a specific error.